test/play-routes-compiler-test.bzl

```diff
@@ -0,0 +1,159 @@
+load("@rules_scala_annex//rules:scala.bzl", "scala_library", "scala_test")
+load("//play-routes:play-routes.bzl", "play_routes")
+
+def generate_play_routes_test_targets(scala_version):
+    # For example 2.13 -> 2-13 or 2_13
+    scala_version_dash = scala_version.replace(".", "-")
+    scala_version_underscore = scala_version.replace(".", "_")
+    scala = "//scala:zinc_{}".format(scala_version_underscore)
+
+    if scala_version == "3":
+        play_routes_toolchain_name = "play-routes-3"
+    elif scala_version == "2.13":
+        play_routes_toolchain_name = "play-routes-2-13"
+    else:
+        fail("Unsupported Scala version when generating test targets")
+
+    play_routes(
+        name = "play-routes-{}".format(scala_version_dash),
+        srcs = ["conf/routes"],
+        include_play_imports = True,
+        play_routes_toolchain_name = play_routes_toolchain_name,
+    )
+
+    play_routes(
+        name = "play-routes-basic1-{}".format(scala_version_dash),
+        srcs = ["conf/basic1.routes"],
+        include_play_imports = True,
+        play_routes_toolchain_name = play_routes_toolchain_name,
+    )
+
+    play_routes(
+        name = "play-routes-basic2-{}".format(scala_version_dash),
+        srcs = ["conf/basic2.routes"],
+        include_play_imports = True,
+        play_routes_toolchain_name = play_routes_toolchain_name,
+    )
+
+    play_routes(
+        name = "play-routes-large-{}".format(scala_version_dash),
+        srcs = ["conf/large.routes"],
+        include_play_imports = True,
+        play_routes_toolchain_name = play_routes_toolchain_name,
+    )
+
+    play_routes(
+        name = "play-routes-additional-imports-{}".format(scala_version_dash),
+        srcs = ["conf/additional_imports.routes"],
+        include_play_imports = True,
+        routes_imports = [
+            "rulesplayroutes.test.User",
+            "rulesplayroutes.test.binders._",
+        ],
+        play_routes_toolchain_name = play_routes_toolchain_name,
+    )
+
+    play_routes(
+        name = "play-routes-specified-generator-{}".format(scala_version_dash),
+        srcs = ["conf/generator.routes"],
+        include_play_imports = True,
+        routes_generator = "play.routes.compiler.InjectedRoutesGenerator",
+        play_routes_toolchain_name = play_routes_toolchain_name,
+    )
+
+    play_routes(
+        name = "play-routes-reverse-router-{}".format(scala_version_dash),
+        srcs = ["conf/reverse_router.routes"],
+        generate_reverse_router = True,
+        include_play_imports = True,
+        play_routes_toolchain_name = play_routes_toolchain_name,
+    )
+
+    # TOOD: Figure out what this does and add the test for it
```

Comment:
Nit: `TODO`

Reply:
This has been here since the dawn of time for this repo. I don't think anyone's ever figured this one out. I'ma leave it as it's just copy pasted from the CLI repo and was written forever ago.

Reply:
Ah, okie dokie. There's no problem preserving relics of the past.